src/hotspot/share/runtime/runtimeUpcallNop.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "runtime/interfaceSupport.inline.hpp"
+#include "runtime/runtimeUpcallNop.hpp"
+#include "runtime/runtimeUpcalls.hpp"
+
+bool runtimeUpcallNop_register_upcalls()
+{
+  if(!AddRuntimeUpcallsNOP) return true;
+  if (RuntimeUpcalls::register_upcall(
+        RuntimeUpcallType::onMethodEntry,
+        "nop_method",
+        RuntimeUpcallNop::nop_method,
+        RuntimeUpcallNop::filter_method_callback)) {
+    return true;
+  }
+  return false;
+}
+
+bool RuntimeUpcallNop::filter_method_callback(MethodDetails& methodDetails)
+{
+  return false;
```

Comment:
Shouldn't this be returning true to enable it for all methods as per the documentation for AddRuntimeUpcallsNOP in globals.hpp?

Reply:
Thanks for pointing that out; I've updated the AddRuntimeUpcallsNOP to a string that is of the format "=[onMethodEntry|onMethodExit]:[all|none]" allowing us to test (a) mulitple upcalls ,(b) test onMethodExit (that it currently asserts), (c) have the NOP upcall do nothing, but either install an upcall on no methods or all methods

Reply:
So now the filter will use the value from the string 'none' (false) or 'all' (true) for the filter result